src/App.tsx

```diff
@@ -42,54 +42,113 @@ export function getReorderedGoods(
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
 
-      <ul>
+export class App extends React.Component {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handleSortClick = (sortType: SortType) => {
+    this.setState({ sortType });
+  }
+
+  handleReverseClick = () => {
+    const { isReversed } = this.state;
+
+    if (isReversed === true) {
+      this.setState({ isReversed: false });
+    } else {
+      this.setState({ isReversed: true });
+    }
+  }
+
+  render() {
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={this.state.sortType === SortType.ALPHABET
+              ? 'button is-info' : 'button is-info is-light'}
+            onClick={() => this.handleSortClick(SortType.ALPHABET)}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={this.state.sortType === SortType.LENGTH
+              ? 'button is-success' : 'button is-success is-light'}
+            onClick={() => this.handleSortClick(SortType.LENGTH)}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={this.state.isReversed === true
+              ? 'button is-warning' : 'button is-warning is-light'}
+            onClick={() => this.handleReverseClick()}
+          >
+            Reverse
+          </button>
+
+          {!(this.state.sortType === SortType.NONE
+              && this.state.isReversed === false)
+            && (
+              <button
+                type="button"
+                className="button is-danger is-light"
+                onClick={() => this
+                  .setState({ isReversed: false, sortType: SortType.NONE })}
+              >
+                Reset
+              </button>
+            )}
+        </div>
+
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            {
+              [...goodsFromServer].sort((good1, good2) => {
```

Comment:
it's a very bad idea to do this here. Use the function provided with the task to achieve sorting and reversing. Before the JSX use this function. 

```
render() {
    const goods = getReorderedGoods(goodsFromServer, this.state)
    return ( 
        // template
    )
}
```

Maybe it's working but it's unreadable and can be better organized.